numpy/f2py/tests/test_abstract_interface.py

```diff
@@ -0,0 +1,65 @@
+import textwrap
+from . import util
+from numpy.f2py import crackfortran
+
+
+class TestAbstractInterface(util.F2PyTest):
+    suffix = '.f90'
+
+    skip = ['add1', 'add2']
+
+    code = textwrap.dedent("""
+        module ops_module
+
+          abstract interface
+            subroutine op(x, y, z)
+              integer, intent(in) :: x, y
+              integer, intent(out) :: z
+            end subroutine
+          end interface
+
+        contains
+
+          subroutine foo(x, y, r1, r2)
+            integer, intent(in) :: x, y
+            integer, intent(out) :: r1, r2
+            procedure (op) add1, add2
+            procedure (op), pointer::p
+            p=>add1
+            call p(x, y, r1)
+            p=>add2
+            call p(x, y, r2)
+          end subroutine
+        end module
+
+        subroutine add1(x, y, z)
+          integer, intent(in) :: x, y
+          integer, intent(out) :: z
+          z = x + y
+        end subroutine
+
+        subroutine add2(x, y, z)
+          integer, intent(in) :: x, y
+          integer, intent(out) :: z
+          z = x + 2 * y
+        end subroutine
+        """)
+
+    def test_abstract_interface(self):
+        assert self.module.ops_module.foo(3, 5) == (8, 13)
+
+    def test_gh18403(self, tmp_path):
```

Comment:
Would be better to make the name more informative, e.g., `test_abstract_interface`.

Reply:
The issue can be put in a comment line.

Reply:
Fixed. Using test_parse_abstract_interface.